chia/data_layer/data_layer.py

```diff
@@ -352,6 +349,19 @@ async def get_kv_diff(self, tree_id: bytes32, hash_1: bytes32, hash_2: bytes32)
         return await self.data_store.get_kv_diff(tree_id, hash_1, hash_2)
 
     async def periodically_fetch_data(self) -> None:
+        success = False
+        while not self._shut_down and not success:
+            async with self.subscription_lock:
+                try:
+                    subscriptions = await self.data_store.get_subscriptions()
+                    for subscription in subscriptions:
+                        await self.wallet_rpc.dl_track_new(subscription.tree_id)
+                    success = True
+                except aiohttp.client_exceptions.ClientConnectorError:
+                    self.log.error("Cannot connect to the wallet. Retrying in 3s.")
+            if not success:
+                await asyncio.sleep(3)
```

Comment:
Three seconds is a pretty long time to delay shutdown, I think we can check more often.  Given that the connection failure is happening regularly on startup I think this should be a warning instead of an error.

```suggestion
        while not self._shut_down:
            async with self.subscription_lock:
                try:
                    subscriptions = await self.data_store.get_subscriptions()
                    for subscription in subscriptions:
                        await self.wallet_rpc.dl_track_new(subscription.tree_id)
                    break
                except aiohttp.client_exceptions.ClientConnectorError:
                    pass

            self.log.warning("Cannot connect to the wallet. Retrying in 3s.")

            delay_until = time.monotonic() + 3
            while time.monotonic() < delay_until:
                if self._shut_down:
                    break
                await asyncio.sleep(0.1)
```